### PR TITLE
Check board_manager_groups when displaying permissions

### PR DESCRIPTION
**Pull request by sbulen**
Now that board managers can see all boards, this needs to be reflected on the 'Show Permissions' view.  

Otherwise, folks with 'Manage Boards and Categories' end up with boards on their restricted boards list, which doesn't make sense.

**_Before change:_**
board_perms_conflict

**_After change:_**
board_perms_conflict-after

Saw this when researching #6450 .

### CODE REVIEW

File: Sources/Profile-View.php [hunk 1]
```diff
@@ -3109,7 +3109,7 @@ function showPermissions($memID)
 	$context['no_access_boards'] = array();
 	while ($row = $smcFunc['db_fetch_assoc']($request))
 	{
-		if (count(array_intersect($curGroups, explode(',', $row['member_groups']))) === 0 && !$row['is_mod'])
+		if ((count(array_intersect($curGroups, explode(',', $row['member_groups']))) === 0) && !$row['is_mod'] && (!empty($modSettings['board_manager_groups']) && count(array_intersect($curGroups, explode(',', $modSettings['board_manager_groups']))) === 0))
```
Comment (albertlast) [inline Sources/Profile-View.php]: to many logic in one line for my pov.

Comment (Sesquipedalian) [inline Sources/Profile-View.php]:
```suggestion
		if (count(array_intersect($curGroups, array_merge(explode(',', $row['member_groups']), explode(',' $modSettings['board_manager_groups'])))) === 0 && !$row['is_mod'])
```

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
Logic is good, but see suggestion for a more parsimonious way to accomplish it.

**Comment by sbulen** [comment]
Recent updates broke this PR...
image

I will simplify the language, retest & resubmit...

**Comment by sbulen** [comment]
I split the logic to make it more readable.
Retested, & tests OK.
I did leave the existing logic, to be consistent with the checks in Load:
https://github.com/SimpleMachines/SMF2.1/blob/6a158f4d4c6d3fb00d165d0537837ef698303b3a/Sources/Load.php#L832

**Review by live627 — APPROVED** [review]

**Pull request merged**